### PR TITLE
fix: Update protocol defs that make bufbuild fail.

### PR DESCRIPTION
**Pull request by richburdon**
- [x] resolves cyclical dependencies.
- [x] first enum value must be ZERO.
- [x] enum names must be unique within package (e.g., add SPACE_ prefix to SpaceState).

### CODE REVIEW

File: packages/core/protocols/src/proto/dxos/client/services.proto [hunk 1]
```diff
@@ -374,7 +379,7 @@ message CreateEpochRequest {
     /// Replace the current automerge root with a new one specified by the user.
     REPLACE_AUTOMERGE_ROOT = 4;
 
-    /// Upgrade references data strucutre 
+    /// Upgrade references data strucutre
```
Comment (wittjosiah) [inline packages/core/protocols/src/proto/dxos/client/services.proto:382]:
```suggestion
    /// Upgrade references data structure
```

File: packages/core/protocols/src/proto/dxos/echo/object.proto [hunk 2]
```diff
@@ -5,13 +5,19 @@
...
+import "google/protobuf/descriptor.proto";
 
 import "dxos/keys.proto";
 import "dxos/halo/credentials.proto";
 import "dxos/echo/timeframe.proto";
 
 package dxos.echo.object;
 
+// TODO(burdon): preserve_any definition.
+extend google.protobuf.FieldOptions {
```
Comment (dmaretskyi) [inline packages/core/protocols/src/proto/dxos/echo/object.proto:17]: extract to `src/proto/field_options.proto`?

**Comment by dmaretskyi** [comment]
Is there a reason this PR is based against a hotfix branch?

**Comment by wittjosiah** [comment]
> Is there a reason this PR is based against a hotfix branch?

we had initially discussed putting it out as a hotfix last week

**Comment by dmaretskyi** [comment]
> > Is there a reason this PR is based against a hotfix branch?
> 
> we had initially discussed putting it out as a hotfix last week

why?

**Comment by wittjosiah** [comment]
> > > Is there a reason this PR is based against a hotfix branch?
> > 
> > 
> > we had initially discussed putting it out as a hotfix last week
> 
> why?

because @richburdon was trying to use it externally so needed them published. After making more substantial changes I believe he changed his mind on that

**Pull request merged**